views/js/previewer/plugins/tools/scale/scale.js

```diff
@@ -0,0 +1,231 @@
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2019 (original work) Open Assessment Technologies SA ;
+ */
+
+/**
+ * Test Previewer Responsive Scale plugin : Scale
+ */
+define([
+    'jquery',
+    'lodash',
+    'i18n',
+    'ui/hider',
+    'taoTests/runner/plugin',
+    'taoQtiTestPreviewer/previewer/utils/devices',
+    'tpl!taoQtiTestPreviewer/previewer/plugins/tools/scale/preview-types',
+    'tpl!taoQtiTestPreviewer/previewer/plugins/tools/scale/mobile-devices',
+    'tpl!taoQtiTestPreviewer/previewer/plugins/tools/scale/desktop-devices',
+    'tpl!taoQtiTestPreviewer/previewer/plugins/tools/scale/scale-wrapper',
+    'taoQtiTestPreviewer/previewer/plugins/tools/scale/responsive-design-testing',
+    'ui/selecter'
+], function (
+    $,
+    _,
+    __,
+    hider,
+    pluginFactory,
+    devices,
+    previewTypesTpl,
+    mobileDevicesTpl,
+    desktopDevicesTpl,
+    scaleWrapperTpl,
+    responsiveDesignTesting,
+    selecter
+) {
+    'use strict';
+
+    return pluginFactory({
+
+        name: 'scale',
+        install: function install(){
+            /**
+             * Tells if the component is enabled
+             * @returns {Boolean}
+             */
+            this.isPluginAllowed = function isPluginAllowed() {
+
+                var config = this.getTestRunner().getConfig();
+                return !config.readOnly;
+            };
+
+            this.api = responsiveDesignTesting(this);
+            /**
+             * device types list to use from "select device type" select-box.
+             * @type {*[]}
+             */
+            this.deviceTypes = [
+                {
+                    value: 'desktop',
+                    label: __('Desktop preview'),
+                    selected: false
+                },
+                {
+                    value: 'mobile',
+                    label: __('Mobile preview'),
+                    selected: false
+                },
+                {
+                    value: 'standard',
+                    label: __('Actual size'),
+                    selected: true
+                },
```

Comment:
Good idea to define the list. But why is it not used in the API? And why is it redefined in several places?
I mean lists with same keys for different meaning are rebuilt on several places. It would have been safer to have the same seed for every ones instead. That would ease future improvements and maintenance.